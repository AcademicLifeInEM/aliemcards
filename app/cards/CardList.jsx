import React, { PropTypes } from 'react';
import { Link } from 'react-router';

const CardList = ({ cards, categories, filter, filterhandler, filtervalue, editortools }) =>
  <div>
    { filter &&
      <select value={filtervalue} onChange={filterhandler}>
        <option value="">Filter by Category:</option>
        {categories.map(category => (
          <option key={category.id} value={category.id}>{category.name}</option>
        ))}
      </select>
    }
    <div className="card-list">
      {cards.map(card =>
        <div key={card.id} className="card-list__item">
          { editortools &&
            <span className="card-list__item-meta ">{ new Date(card.created).toLocaleDateString('en-US') }</span>
          }
          <Link to={`/cards/${card.id}`} className="card-list__item-title">{card.title}</Link>{' '}
          {card.categories.map(category =>
            <Link to={`/categories/${category.id}`} key={`${card.id}-${category.id}`} className="card-list__item-meta">{category.name.toLowerCase()}</Link>
          )}
        </div>
      )}
    </div>
  </div>;

CardList.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    categories: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
    })),
  })),
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  })),
  editortools: PropTypes.bool,
  filter: PropTypes.bool,
  filtervalue: PropTypes.string,
  filterhandler: PropTypes.func,
};

export default CardList;
