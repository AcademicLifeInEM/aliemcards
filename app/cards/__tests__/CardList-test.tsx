jest.mock('react-router-dom');

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CardList from '../CardList';

const data: any = {
  cards: [
    {
      id: '1',
      categories: [
        {
          id: '1',
          name: 'Test Category',
        },
      ],
      title: 'Test Card',
    },
  ],
  categories: [
    {
      id: '1',
      name: 'Test Category',
    },
  ],
};

const setup = (filter: boolean = false) => {
  const props = { ...data };
  const snapshot = renderer.create(
    <CardList
      cards={props.cards}
      categories={props.categories}
      filterhandler={jest.fn()}
      filter={filter}
      filtervalue=""
    />,
  );
  return {
    snapshot,
  };
};

describe('<CardList />', () => {
  it('should render without filter', () => {
    const { snapshot } = setup();
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
  it('should render with filter', () => {
    const { snapshot } = setup(true);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
