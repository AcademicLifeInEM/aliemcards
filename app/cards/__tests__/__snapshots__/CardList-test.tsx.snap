// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render with filter 1`] = `
<div>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      Filter by Category:
    </option>
    <option
      value="1"
    >
      Test Category
    </option>
  </select>
  <div
    className="card-list"
  >
    <div
      className="card-list__item"
    >
       
    </div>
  </div>
</div>
`;

exports[`<CardList /> should render without filter 1`] = `
<div>
  <div
    className="card-list"
  >
    <div
      className="card-list__item"
    >
       
    </div>
  </div>
</div>
`;
